osmomath/decimal_test.go

```diff
@@ -739,3 +739,282 @@ func (s *decimalTestSuite) TestLog2() {
+		},
+		"log_e{1} = 0": {
+			initialValue: NewBigDec(1),
+			expected:     NewBigDec(0),
+		},
+		"log_e{e} = 1": {
+			initialValue: MustNewDecFromStr("2.718281828459045235360287471352662498"),
+			// From: https://www.wolframalpha.com/input?i=e+with+36+decimals
+			expected: NewBigDec(1),
+		},
+		"log_e{7} = 1.945910149055313305105352743443179730": {
+			initialValue: NewBigDec(7),
+			// From: https://www.wolframalpha.com/input?i=log7+up+to+36+decimals
+			expected: MustNewDecFromStr("1.945910149055313305105352743443179730"),
+		},
+		"log_e{512} = 6.238324625039507784755089093123589113": {
+			initialValue: NewBigDec(512),
+			// From: https://www.wolframalpha.com/input?i=log512+up+to+36+decimals
+			expected: MustNewDecFromStr("6.238324625039507784755089093123589113"),
+		},
+		"log_e{580} = 6.36302810354046502061849560850445238": {
+			initialValue: NewBigDec(580),
+			// From: https://www.wolframalpha.com/input?i=log580+up+to+36+decimals
+			expected: MustNewDecFromStr("6.36302810354046502061849560850445238"),
+		},
+		"log_e{1024.987654321} = 6.93243584693509415029056534690631614": {
+			initialValue: NewDecWithPrec(1024987654321, 9),
+			// From: https://www.wolframalpha.com/input?i=log1024.987654321+to+36+digits
+			expected: MustNewDecFromStr("6.93243584693509415029056534690631614"),
+		},
+		"log_e{912648174127941279170121098210.92821920190204131121} = 68.986147965719214790400745338243805015": {
+			initialValue: MustNewDecFromStr("912648174127941279170121098210.92821920190204131121"),
+			// From: https://www.wolframalpha.com/input?i=log912648174127941279170121098210.92821920190204131121+to+38+digits
+			expected: MustNewDecFromStr("68.986147965719214790400745338243805015"),
+		},
+	}
+
+	for name, tc := range tests {
+		s.Run(name, func() {
+			osmoassert.ConditionalPanic(s.T(), tc.expectedPanic, func() {
+				// Create a copy to test that the original was not modified.
+				// That is, that Ln() is non-mutative.
+				initialCopy := ZeroDec()
+				initialCopy.i.Set(tc.initialValue.i)
+
+				// system under test.
+				res := tc.initialValue.Ln()
+				require.True(DecApproxEq(s.T(), tc.expected, res, expectedErrTolerance))
+				require.Equal(s.T(), initialCopy, tc.initialValue)
+			})
+		})
+	}
+}
+
+func (s *decimalTestSuite) TestTickLog() {
+	tests := map[string]struct {
+		initialValue BigDec
+		expected     BigDec
+
+		expectedErrTolerance BigDec
+		expectedPanic        bool
+	}{
+		"log_1.0001{-1}; invalid; panic": {
+			initialValue:  OneDec().Neg(),
+			expectedPanic: true,
+		},
+		"log_1.0001{0}; invalid; panic": {
+			initialValue:  ZeroDec(),
+			expectedPanic: true,
+		},
+		"log_1.0001{0.001} = -69081.006609899112313305835611219486392199": {
+			initialValue: MustNewDecFromStr("0.001"),
+			// From: https://www.wolframalpha.com/input?i=log_1.0001%280.001%29+to+41+digits
+			expectedErrTolerance: MustNewDecFromStr("0.000000000000000000000000000143031879"),
```

Comment:
Do we think it makes more sense to have varying error tolerance levels for each test case or just a single overarching one for all the tests? The later seems more readable but I could be missing some important reason why this is not being done.

Reply:
While I agree that the latter is more readable, I think it's helpful to know the error tolerance for some subsets of inputs that we can infer from test cases.

I would prefer this knowledge over readability here